// @storefront-ui/vue/dist/components/SfRatingButton/SfRatingButton.vue.mjs
import { defineComponent as B, toRefs as $, ref as F, computed as y, openBlock as o, createElementBlock as u, Fragment as R, renderList as _, withDirectives as I, createElementVNode as L, unref as n, vModelRadio as N, renderSlot as E, createBlock as x } from "vue";
import H from "../SfIcons/SfIconStarFilled.vue.mjs";
import T from "../SfIcons/SfIconStar.vue.mjs";
import { SfRatingButtonSize as i, SfIconSize as c } from "@storefront-ui/shared";
const w = {
  role: "radiogroup",
  class: "flex",
  "data-testid": "ratingbutton"
}, A = ["onMouseenter"], C = ["name", "value", "disabled", "aria-label"], j = /* @__PURE__ */ B({
  __name: "SfRatingButton",
  props: {
    modelValue: {
      type: Number,
      default: 0
    },
    max: {
      type: Number,
      default: 5
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    name: {
      type: String,
      default: "sf-rating-button"
    },
    size: {
      type: String,
      default: i.base
    },
    getLabelText: {
      type: Function,
      default: (e) => `${e} Star${e !== 1 ? "s" : ""}`
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: z }) {
    const m = e, S = z, { max: f, disabled: r, modelValue: h } = $(m), a = F(0), g = y(() => Array.from({ length: Math.floor(Math.abs(f.value)) }, (t, s) => s + 1)), b = (t) => t <= a.value || a.value === 0 && t <= h.value, d = {
      [i.sm]: c.base,
      [i.base]: c.lg,
      [i.lg]: c.xl
    }, p = y({
      set(t) {
        S("update:modelValue", t);
      },
      get() {
        return m.modelValue;
      }
    }), M = (t) => {
      r.value || (a.value = t);
    }, k = () => {
      r.value || (a.value = 0);
    };
    return (t, s) => (o(), u("div", w, [
      (o(!0), u(R, null, _(g.value, (l) => (o(), u("label", {
        key: l,
        onMouseenter: (v) => M(l),
        onMouseleave: k
      }, [
        I(L("input", {
          "onUpdate:modelValue": s[0] || (s[0] = (v) => p.value = v),
          type: "radio",
          class: "sr-only peer",
          name: e.name,
          value: l,
          disabled: n(r),
          "aria-label": e.getLabelText(l)
        }, null, 8, C), [
          [N, p.value]
        ]),
        E(t.$slots, "default", {
          isFilled: b(l),
          iconSize: d[e.size],
          max: n(f)
        }, () => [
          b(l) ? (o(), x(n(H), {
            key: 0,
            role: "none",
            class: "text-primary-700 cursor-pointer peer-disabled:cursor-default peer-disabled:text-disabled-500 peer-focus-visible:outline",
            size: d[e.size]
          }, null, 8, ["size"])) : (o(), x(n(T), {
            key: 1,
            role: "none",
            class: "text-neutral-500 cursor-pointer peer-disabled:cursor-default peer-disabled:text-disabled-500 peer-focus-visible:outline",
            size: d[e.size]
          }, null, 8, ["size"]))
        ])
      ], 40, A))), 128))
    ]));
  }
});
export {
  j as default
};
