// @storefront-ui/vue/dist/components/SfLoaderLinear/SfLoaderLinear.vue.mjs
import { defineComponent as r, openBlock as l, createElementBlock as t, normalizeClass as i } from "vue";
import { SfLoaderLinearSize as n } from "./types.mjs";
import { SfLoaderSize as e } from "@storefront-ui/shared";
const o = ["aria-label"], s = {
  [n.minimal]: "h-1",
  [e.xs]: "h-4",
  [e.sm]: "h-5",
  [e.base]: "h-6",
  [e.lg]: "h-7",
  [e.xl]: "h-10",
  [e["2xl"]]: "h-14",
  [e["3xl"]]: "h-24",
  [e["4xl"]]: "h-48"
}, p = /* @__PURE__ */ r({
  __name: "SfLoaderLinear",
  props: {
    size: {
      type: String,
      default: e.base
    },
    ariaLabel: {
      type: String,
      default: "loading"
    }
  },
  setup(a) {
    return (f, m) => (l(), t("span", {
      class: i(["relative inline-block overflow-hidden bg-neutral-300 text-primary-700 after:absolute after:w-2.5 after:h-full after:animate-line after:bg-current after:block", s[a.size]]),
      "aria-live": "polite",
      "aria-label": a.ariaLabel,
      "data-testid": "loader-linear"
    }, null, 10, o));
  }
});
export {
  p as default
};
