// @storefront-ui/vue/dist/components/SfProgressLinear/SfProgressLinear.vue.mjs
import { defineComponent as a, openBlock as s, createElementBlock as t, normalizeClass as i } from "vue";
import { SfProgressLinearSize as l } from "./types.mjs";
import { SfProgressSize as e } from "@storefront-ui/shared";
const o = ["aria-label", "value"], n = {
  [l.minimal]: "h-1",
  [e.xs]: "h-4",
  [e.sm]: "h-5",
  [e.base]: "h-6",
  [e.lg]: "h-7",
  [e.xl]: "h-10",
  [e["2xl"]]: "h-14",
  [e["3xl"]]: "h-24",
  [e["4xl"]]: "h-48"
}, c = /* @__PURE__ */ a({
  __name: "SfProgressLinear",
  props: {
    value: {
      type: Number,
      default: 0
    },
    size: {
      type: String,
      default: e.base
    },
    ariaLabel: {
      type: String,
      default: "Progress element"
    }
  },
  setup(r) {
    return (u, g) => (s(), t("progress", {
      "data-testid": "progress-linear",
      max: "100",
      class: i(["bg-neutral-300 text-primary-700 [&::-webkit-progress-bar]:bg-inherit [&::-webkit-progress-value]:bg-current [&::-webkit-progress-value]:transition-[width] [&::-webkit-progress-value]:ease-in-out [&::-webkit-progress-value]:duration-200 [&::-moz-progress-bar]:bg-current", n[r.size]]),
      "aria-label": r.ariaLabel,
      value: r.value
    }, null, 10, o));
  }
});
export {
  c as default
};
