// @vue-storefront/rollup-config/lib/base.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateBaseConfig = void 0;
const plugin_node_resolve_1 = __importDefault(require("@rollup/plugin-node-resolve"));
const rollup_plugin_typescript2_1 = __importDefault(require("rollup-plugin-typescript2"));
const extensions = ['.ts', '.js'];
function generateBaseConfig(pkg) {
    return {
        input: 'src/index.ts',
        output: [
            {
                file: pkg.main,
                format: 'cjs',
                sourcemap: true
            },
            {
                file: pkg.module,
                format: 'es',
                sourcemap: true
            }
        ],
        external: [
            ...Object.keys(pkg.dependencies || {}),
            ...Object.keys(pkg.peerDependencies || {})
        ],
        plugins: [
            (0, plugin_node_resolve_1.default)({
                extensions
            }),
            (0, rollup_plugin_typescript2_1.default)(),
        ]
    };
}
exports.generateBaseConfig = generateBaseConfig;
